File: Traffic-Transfer-Demo/Traffic-Transfer-Demo/TrafficTransferCollectionViewCell.m
//
//  TrafficTransferCollectionViewCell.m
//  Traffic-Transfer-Demo
//
//  Created by eidan on 16/11/29.
//  Copyright © 2016年 autonavi. All rights reserved.
//

#import "TrafficTransferCollectionViewCell.h"
#import <AMapSearchKit/AMapSearchKit.h>

@interface TrafficTransferCollectionViewCell ()

@property (weak, nonatomic) IBOutlet UILabel *trafficTransferInfoLabel;
@property (weak, nonatomic) IBOutlet UILabel *otherInfoLabel;


@end

@implementation TrafficTransferCollectionViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)configureWithTransit:(AMapTransit *)transit {
    
    NSMutableArray *buslineArray = @[].mutableCopy;
    for (AMapSegment *segment in transit.segments) {
        AMapRailway *railway = segment.railway; //火车,一个城市内的不会出现火车。
        AMapBusLine *busline = [segment.buslines firstObject];  // 地铁或者公交线路
        if (busline.name) {
            [buslineArray addObject:busline.name];
        } else if (railway.uid) {
            [buslineArray addObject:railway.name];
        }
    }
    self.trafficTransferInfoLabel.text = [buslineArray componentsJoinedByString:@" > "];
    
    NSInteger hours = transit.duration / 3600;
    NSInteger minutes = (NSInteger)(transit.duration / 60) % 60;
    self.otherInfoLabel.text = [NSString stringWithFormat:@"%u小时%u分钟 | %u公里 | 步行%.1f公里",(unsigned)hours,(unsigned)minutes,(unsigned)transit.distance / 1000,transit.walkingDistance / 1000.0];
}


@end

File: Traffic-Transfer-Demo-Swift/TrafficTransferCollectionViewCell.swift
//
//  TrafficTransferCollectionViewCell.swift
//  Traffic-Transfer-Demo
//
//  Created by eidan on 17/1/6.
//  Copyright © 2017年 autonavi. All rights reserved.
//

import UIKit

class TrafficTransferCollectionViewCell: UICollectionViewCell {
    
    @IBOutlet weak var trafficTransferInfoLabel: UILabel!

    @IBOutlet weak var otherInfoLabel: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    func configureWithTransit(transit: AMapTransit ) {
        
        var buslineArray = [String]();
        
        for segment: AMapSegment in transit.segments {
            let railway = segment.railway
            let busline = segment.buslines.first
            if busline?.name != nil {
                buslineArray.append((busline?.name)!)
            } else if ((railway?.uid) != nil) {
                buslineArray.append((railway?.name)!)
            }
        }
        
        self.trafficTransferInfoLabel.text = buslineArray.joined(separator: " > ")
        
        let hours = transit.duration / 3600
        let minutes = (transit.duration / 60) % 60
        let distance = String(format: "%.0f", CGFloat(transit.distance) / CGFloat(1000))
        let walkDistance = String(format: "%0.1f", CGFloat(transit.walkingDistance) / CGFloat(1000))
        let cost = String(format: "%.0f", transit.cost)
        self.otherInfoLabel.text = "\(hours)小时\(minutes)分钟 | \(distance)公里 | \(cost)元 | 步行\(walkDistance)公里"
    }
    
    

}
